hyper/http11/connection.py

```diff
@@ -93,11 +105,16 @@ def connect(self):
         :returns: Nothing.
         """
         if self._sock is None:
-            sock = socket.create_connection((self.host, self.port), 5)
+            if not self.proxy_host:
+                host = self.host
+            else:
+                port = self.proxy_host
```

Comment:
This is not quite right: you assign only host in the top block and only port in the bottom block. Need to do both in both. =)